client/src/components/App.jsx

```diff
@@ -31,19 +34,28 @@ class App extends React.Component {
       data: initialState,
       imageList: false,
       clickedPhoto: 0,
+      is_favorite: 0,
     }
     this.getRoomData = this.getRoomData.bind(this);
     this.toggleMainAndPhotoList = this.toggleMainAndPhotoList.bind(this);
+    this.toggleFavorite = this.toggleFavorite.bind(this);
+    this.updateFavorite = this.updateFavorite.bind(this);
   }
 
   toggleMainAndPhotoList(clickedPhoto) {
-    console.log(clickedPhoto, 'in APP');
     this.setState({
       clickedPhoto: clickedPhoto,
       imageList: !this.state.imageList,
     })
   }
 
+  toggleFavorite() {
+    this.setState({
+      is_favorite: !this.state.is_favorite,
```

Comment:
Yes, it is because of setState. In the scope of toggleFavorite, both setState and updateFavorite are executed, but because there are executing in the same scope still, patch request to updateFavorite is still executed with old state if state is not updated with setState. So, because of the scope and sync nature. (hmm.. hope it makes sense)

Reply:
@tegsheeb Ok, just wanted to make sure. That's kinda what I figured but wanted to check.